registration/views.py

```diff
@@ -231,6 +231,17 @@ def get(self, request, *args, **kwargs):
         clean_session(request.session)
         clean_next_of_kin_email_session(request.session)
         setup_external_source_session(request)
+        if self.request.GET.get('email') and self.request.GET.get('postal_code'):
+            email = self.request.GET['email']
+            postal_code = self.request.GET['postal_code']
+            # Email and postal codes are provided, go ahead and process them
+            try:
+                self.submit_email({'email': email, 'postal_code': postal_code})
```

Comment:
Actually the current behavior is setting initial using `get_initial()`. We are using `get_form()` to instantiate the form in which is uses the following code:
```
    def get_form(self, form_class=None):
        """
        Returns an instance of the form to be used in this view.
        """
        if form_class is None:
            form_class = self.get_form_class()
        return form_class(**self.get_form_kwargs())

    def get_form_kwargs(self):
        """
        Returns the keyword arguments for instantiating the form.
        """
        kwargs = {
            'initial': self.get_initial(),
            'prefix': self.get_prefix(),
        }

        if self.request.method in ('POST', 'PUT'):
            kwargs.update({
                'data': self.request.POST,
                'files': self.request.FILES,
            })
        return kwargs
```
As you can see, its the same as your suggestion.
The problem is that the `is_valid()` method check if the from is `is_bound` - check the its source code below:
```
    def is_valid(self):
        """
        Returns True if the form has no errors. Otherwise, False. If errors are
        being ignored, returns False.
        """
        return self.is_bound and not self.errors
```
In the form constructor, `is_bound` is set to `True` only if `data` is provided in the constructor, its not respecting the `initial`
```
def __init__(self, data=None, files=None, auto_id='id_%s', prefix=None,
                 initial=None, error_class=ErrorList, label_suffix=None,
                 empty_permitted=False, field_order=None):
        self.is_bound = data is not None or files is not None
        self.data = data or {}
        self.files = files or {}
        self.auto_id = auto_id
        if prefix is not None:
            self.prefix = prefix
        self.initial = initial or {}
```